log10/load.py

```diff
@@ -246,28 +246,50 @@ def wrapper(*args, **kwargs):
                 }
                 for frame in current_stack_frame
             ]
+            assert (
+                kwargs is not None
+            ), "Get empty request. Please check the input parameters and try again."
 
             start_time = time.perf_counter()
             output = func_with_backoff(func, *args, **kwargs)
             duration = time.perf_counter() - start_time
             logging.debug(f"LOG10: TIMED BLOCK - LLM call duration: {duration}")
-
+        except Exception as e:
+            if USE_ASYNC:
+                with timed_block("extra time spent waiting for log10 call"):
+                    while result_queue.empty():
+                        pass
+                    completionID = result_queue.get()
+            logging.debug("LOG10: failed", e)
+            log_row = {
+                "status": "failed",
+                "failure_reason": str(e),
```

Comment:
Could we add a "failure_kind" field with the common error types we know of now?

 - Time out
 - Rate limit
 - Auth issue
 - Max tokens exceeded
 - Unknown

Reply:
yes, I can add the error type `type(e).__name__`. 
I could combine the failure_reason and error type together, like:
``` python
reason = {"message": str(e), "kind": type(e).__name__}
log_row = {
    "failure_reason": reason,
}
```

or simply concat both into one message:
``` python
"failure_reason": f"Got {type(e).__name__}. Message: {str(e)}"
```

Any preference? Which one the platform would like to use?

Reply:
cool, I added a the new field "failure_kind".

Reply:
@nqn please review.

Reply:
Yeah, we'd like two field - one with the whole error message and one which reduce it into one of a few high level categories i.e. kinds.

The context is being able to do stats and alerting easily, if there's a high concentration of, say, time outs, exceeded max lenghts etc.

We were debating whether to do this translation in the API or in the library, but think the library has more context about the errors, so that it'd make sense to do there.

For high level categories, thought of:

 - Time out
 - Rate limit
 - Auth issues
 - Max tokens exceeded
 - Invalid arguments (parameters)
 - Unknown error


Does that make sense? If not, we can maybe huddle a bit about it

Reply:
Openai errors covers most of the high level categories, but used `InvalidRequestError` (`BadRequestError` in [openai-python v1](https://platform.openai.com/docs/guides/error-codes/python-library-error-types)) for both "max tokens exceeded" and "invalid arguments" errors. The max tokens exceeded error message is like this: 

> openai.error.InvalidRequestError: This model's maximum context length is 4097 tokens, however you requested 4118 tokens (22 in your prompt; 4096 for the completion). Please reduce your prompt; or completion length.

**Latest commit**: based on the error type and message, we set a new`ContextWindowExceedError` as the failure_kind along with the original message, to differentiate from other `InvalidRequestError`.

This part needs to be updated when upgrading `openai-python` lib to v1+. 

Note: We could apply similar method to filter the error from Anthropic, but Anthropic has a 100k+ token limit for claude.